### PR TITLE
Upgrade gatsby to v2

### PR DESCRIPTION
**Pull request by RobertHebel**
Update Gatsby and its plugins to v2.
Unfortunately, some libs weren't updated to the latest version because they break current functionalities

### CODE REVIEW

File: src/components/footer/Footer.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import styles from './footer.module.scss';
 
 const Footer = () => (
   <footer className={styles.footer}>
-    <p>&copy; 2018 Maintained by <a href="https://brainhub.eu">Brainhub</a></p>
+    <p>&copy; {new Date().getFullYear()} Maintained by <a href="https://brainhub.eu">Brainhub</a></p>
```
Comment (piotr-s-brainhub) [inline src/components/footer/Footer.js]: I'm not sure whether the year should change

File: docs-www/gatsby-config.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ module.exports = {
   },
 
   // URL prefix on production environment. For more info see https://www.gatsbyjs.org/docs/path-prefix/
-  pathPrefix: process.env.PATH_PREFIX || ' ',
+  pathPrefix: process.env.PATH_PREFIX || '',
```
Comment (piotr-s-brainhub) [inline docs-www/gatsby-config.js]: in carousel the default is null
https://github.com/brainhubeu/react-carousel/blob/master/docs-www/gatsby-config.js#L24

so what's better?

File: src/components/footer/Footer.js (hunk 1 above)
Comment (RobertHebel) [reply src/components/footer/Footer.js]: Why? In the same way, the year is set up on our website

Comment (piotr-s-brainhub) [reply src/components/footer/Footer.js]: ok

File: docs-www/gatsby-config.js (hunk 2 above)
Comment (RobertHebel) [reply docs-www/gatsby-config.js]: In both cases, pathPrefix has falsy value for default. Applying them end up with the same result. Idk which option is better.

Comment (piotr-s-brainhub) [reply docs-www/gatsby-config.js]: ok

**Pull request merged**